provider/core/src/serde.rs

```diff
@@ -280,46 +280,46 @@ impl<'a> Deref for SerdeSeDataStructDynRef<'a> {
     }
 }
 
-impl<'data, M> crate::dynutil::UpcastDataPayload<'data, M> for SerdeSeDataStructMarker
+impl<M> crate::dynutil::UpcastDataPayload<M> for SerdeSeDataStructMarker
 where
     M: DataMarker,
     for<'a> &'a <M::Yokeable as Yokeable<'a>>::Output: serde::Serialize,
 {
-    fn upcast(other: DataPayload<'data, M>) -> DataPayload<'data, SerdeSeDataStructMarker> {
+    fn upcast(other: DataPayload<M>) -> DataPayload<SerdeSeDataStructMarker> {
         use crate::data_provider::{DataPayloadInner, ErasedCart};
         let yoke: Yoke<_, ErasedCart> = match other.inner {
             DataPayloadInner::RcStruct(yoke) => {
                 let rc: Rc<Yoke<_, _>> = Rc::from(yoke);
                 let yoke =
-                    Yoke::attach_to_rc_cart(rc.clone() as Rc<dyn SerdeSeDataStruct<'data> + 'data>);
+                    Yoke::attach_to_rc_cart(rc.clone() as Rc<dyn SerdeSeDataStruct<'static>>);
```

Comment:
Request: Try to see if it's possible to get rid of the lifetime on `SerdeSeDataStruct`.

Reply:
I plan to do this immediately in a followup, this change was not necessary to land the rest of this so I didn't do it.